pkg/querier/metadata_handler.go

```diff
@@ -46,17 +47,47 @@ type metadataErrorResult struct {
 // Mimir for a given tenant. It is kept and returned as a set.
 func NewMetadataHandler(m MetadataSupplier) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		limit := -1
+		if s := r.FormValue("limit"); s != "" {
+			var err error
+			if limit, err = strconv.Atoi(s); err != nil {
+				w.WriteHeader(http.StatusBadRequest)
+				util.WriteJSONResponse(w, metadataErrorResult{Status: statusError, Error: "limit must be a number"})
+				return
+			}
+		}
+		limitPerMetric := -1
+		if s := r.FormValue("limit_per_metric"); s != "" {
+			var err error
+			if limitPerMetric, err = strconv.Atoi(s); err != nil {
+				w.WriteHeader(http.StatusBadRequest)
+				util.WriteJSONResponse(w, metadataErrorResult{Status: statusError, Error: "limit_per_metric must be a number"})
+				return
+			}
+		}
+
 		resp, err := m.MetricsMetadata(r.Context())
 		if err != nil {
 			w.WriteHeader(http.StatusBadRequest)
 			util.WriteJSONResponse(w, metadataErrorResult{Status: statusError, Error: err.Error()})
 			return
 		}
 
+		metric := r.FormValue("metric")
```

Comment:
Similar to above - would it be possible to send the metric to ingesters and store-gateways, so they can filter their response?